ydb/query/transaction.py

```diff
@@ -0,0 +1,369 @@
+import abc
+import logging
+import enum
+import functools
+
+from .. import (
+    _apis,
+    issues,
+)
+from .._grpc.grpcwrapper import ydb_query as _ydb_query
+from .._grpc.grpcwrapper import ydb_query_public_types as _ydb_query_public
+
+from . import base
+
+logger = logging.getLogger(__name__)
+
+
+class QueryTxStateEnum(enum.Enum):
+    NOT_INITIALIZED = "NOT_INITIALIZED"
+    BEGINED = "BEGINED"
+    COMMITTED = "COMMITTED"
+    ROLLBACKED = "ROLLBACKED"
+    DEAD = "DEAD"
+
+
+class QueryTxStateHelper(abc.ABC):
+    _VALID_TRANSITIONS = {
+        QueryTxStateEnum.NOT_INITIALIZED: [
+            QueryTxStateEnum.BEGINED,
+            QueryTxStateEnum.DEAD,
+            QueryTxStateEnum.COMMITTED,
+            QueryTxStateEnum.ROLLBACKED,
+        ],
+        QueryTxStateEnum.BEGINED: [QueryTxStateEnum.COMMITTED, QueryTxStateEnum.ROLLBACKED, QueryTxStateEnum.DEAD],
+        QueryTxStateEnum.COMMITTED: [],
+        QueryTxStateEnum.ROLLBACKED: [],
+        QueryTxStateEnum.DEAD: [],
+    }
+
+    _TERMINAL_STATES = [
```

Comment:
Do you need explicit terminal state?
What if use `len(_VALID_TRANSITIONS[state]) == 0` for detect terminal states?